LongWarOfTheChosen/Src/LW_PerkPack_Integrated/Classes/X2Effect_GreaterPadding.uc

```diff
@@ -0,0 +1,124 @@
+///---------------------------------------------------------------------------------------
+//  FILE:    X2Effect_GreaterPadding
+//  AUTHOR:  Grobobobo (taken from field surgeon)
+//  PURPOSE: Implements effect for FieldSurgeon ability -- this allows healing of lowest hp at end of mission
+//--------------------------------------------------------------------------------------- 
+//---------------------------------------------------------------------------------------
+class X2Effect_GreaterPadding extends X2Effect_Persistent config(LW_SoldierSkills);
+
+var name GreaterPaddingUnitWasBleedingOut;
+var int Padding_HealHP;
+
+function RegisterForEvents(XComGameState_Effect EffectGameState)
+{
+	local Object EffectObj;
+
+	EffectObj = EffectGameState;
+
+	// Remove the default UnitRemovedFromPlay registered by XComGameState_Effect. This is necessary so we can
+	// suppress the usual behavior of the effects being removed when a unit evacs. We can't process field surgeon
+	// at that time because we could evac a wounded unit and then have the surgeon get killed on a later turn. We
+	// need to wait until the mission ends and then process FS.
+	`XEVENTMGR.UnRegisterFromEvent(EffectObj, 'UnitRemovedFromPlay');
+
+	// Because bleeding out status will be cleared at the CleanupTacticalMission point we have to record
+	// that unit was bleeding out at some point during the mission.
+	`XEVENTMGR.RegisterForEvent(EffectObj, 'UnitBleedingOut', OnUnitBleedingOut, ELD_OnStateSubmitted, ,,, EffectObj);
+}
+
+function ApplyGreaterPadding(XComGameState_Effect EffectState, XComGameState_Unit OrigUnitState, XComGameState NewGameState)
+{
+	local XComGameState_Unit		SourceUnitState, UnitState;
+	local UnitValue					StatusValue;
+	local int						StatusIntValue;
+
+	UnitState = XComGameState_Unit(NewGameState.GetGameStateForObjectID(OrigUnitState.ObjectID));
+	if (UnitState == none)
+	{
+		UnitState = XComGameState_Unit(NewGameState.CreateStateObject(class'XComGameState_Unit', OrigUnitState.ObjectID));
+		NewGameState.AddStateObject(UnitState);
+	}
+
+	SourceUnitState = XComGameState_Unit(`XCOMHISTORY.GetGameStateForObjectID(EffectState.ApplyEffectParameters.SourceStateObjectRef.ObjectID));
+
+	if(!GreaterPaddingEffectIsValidForSource(SourceUnitState)) { return; }
+
+	if(UnitState == none) { return; }
+	if(UnitState.IsDead()) { return; }
+	// This check does nothing, as it is seemingly always false at this point. 
+	// Leaving it here just in case.
+	if(UnitState.IsBleedingOut()) { return; }
+
+	if(UnitState.GetUnitValue(default.GreaterPaddingUnitWasBleedingOut, StatusValue))
+	{		
+		StatusIntValue = int(StatusValue.fValue);
+		if(StatusIntValue > 0) { return; }
+	}
+
+
+	if(!CanBeHealed(UnitState)) { return; }
+
+    UnitState.LowestHP += Min(UnitState.HighestHP-UnitState.LowestHP,Padding_HealHP);
```

Comment:
```suggestion
	if (!GreaterPaddingEffectIsValidForSource(SourceUnitState)) { return; }

	if (UnitState == none) { return; }
	if (UnitState.IsDead()) { return; }
	// This check does nothing, as it is seemingly always false at this point. 
	// Leaving it here just in case.
	if (UnitState.IsBleedingOut()) { return; }

	if (UnitState.GetUnitValue(default.GreaterPaddingUnitWasBleedingOut, StatusValue))
	{		
		StatusIntValue = int(StatusValue.fValue);
		if(StatusIntValue > 0) { return; }
	}

	if (!CanBeHealed(UnitState)) { return; }

	UnitState.LowestHP += Min(UnitState.HighestHP-UnitState.LowestHP, Padding_HealHP);
```